### PR TITLE
Restructure ArgoCD app to improve sync performance

### PR DESCRIPTION
**Pull request by anishasthana**
The ArgoCD app was previously reponsible for keeping both itself and
the existence all managed applications in sync, resulting in syncs
taking a long time to complete. This commit separates out applications
into their own 'argocd-apps" application so that sync times are improved both for
ArgoCD itself as well as child apps

Signed-off-by: Anish Asthana <anishasthana1@gmail.com>

### CODE REVIEW

File: manifests/overlays/prod/applications/argocd/aicoe-argocd.yaml [hunk 1]
```diff
@@ -16,4 +16,4 @@ spec:
       prune: true
       selfHeal: true
     syncOptions:
-    - Validate=false
+      - Validate=false
```
Comment (accorvin) [inline manifests/overlays/prod/applications/argocd/aicoe-argocd.yaml:19]: Should this be a separate commit?

Comment (anishasthana) [reply manifests/overlays/prod/applications/argocd/aicoe-argocd.yaml]: Fixed.

**Review by harshad16 — APPROVED** [review]
lgtm

**Review by accorvin — APPROVED** [review]

**Review by HumairAK — APPROVED** [review]

File: OWNERS [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
   - anishasthana
+  - gmfrasca
   - HumairAK
-  - tumido
+  - maulikjs
+  - rimolive
   - sesheta
+  - tumido
 reviewers:
-  - durandom
```
Comment (tumido) [inline OWNERS:9]: @anishasthana I think you want us in the reviewers list as well. As far as I understand on how this works, `prow`  requests review from "somebody" from the `reviewers` list. So if we don't want all the reviews to be assigned to @harshad16 , we should ourselves in here as well.

Comment (anishasthana) [reply OWNERS:9]: Ah.. Got it.
